File: Atcoder_#361/c/b_intersection_of_cuboids.c
#include <stdio.h>

int max(int a, int b) {
    return (a > b) ? a : b;
}

int min(int a, int b) {
    return (a < b) ? a : b;
}

int main() {
    int a,b,c,d,e,f;
    scanf("%d", &a);
    scanf("%d", &b);
    scanf("%d", &c);
    scanf("%d", &d);
    scanf("%d", &e);
    scanf("%d", &f);

    int g,h,i,j,k,l;
    scanf("%d", &g);
    scanf("%d", &h);
    scanf("%d", &i);
    scanf("%d", &j);
    scanf("%d", &k);
    scanf("%d", &l);

    int x = max(0, min(d, j) - max(a, g));
    int y = max(0, min(e, k) - max(b, h));
    int z = max(0, min(f, l) - max(c, i));

    int intersectVolume = x * y * z;

    if (intersectVolume > 0) {
        printf("Yes\n");
    } else {
        printf("No\n");
    }

    return 0;
}

File: Atcoder_#361/javascript/b_intersection_of_cuboids.js
function main(input) {
  const lines = input.split("\n");
  const [a,b,c,d,e,f] = lines[0].split(' ').map(Number);
  const [g,h,i,j,k,l] = lines[1].split(' ').map(Number);

  const x = Math.max(0, Math.min(d, j) - Math.max(a, g));
  const y = Math.max(0, Math.min(e, k) - Math.max(b, h));
  const z = Math.max(0, Math.min(f, l) - Math.max(c, i));

  const ans = x * y * z
  const result = ans > 0 ? "Yes" : "No";
  console.log(result);

}
main(require("fs").readFileSync("/dev/stdin", "utf8"));
